### PR TITLE
SALTO-5735: Add requestType field in automaiton

### PR DESCRIPTION
**Pull request by idoamit8**
In this PR I modified the requestType field in automation to be a reference and added required `serviceDesk` field with the correct value.  

---

_Additional context for reviewer_

---
_Release Notes_: 
None

---
_User Notifications_: 
_Jira Adapter:_
* for automation nacls: serviceDesk field will be removed and requestType field will be converted from string to reference expression.

### CODE REVIEW

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 1]
```diff
@@ -120,9 +120,28 @@ const ASSET_COMPONENT_SCHEME = Joi.object({
   })
     .unknown(true)
     .required(),
+})
+  .unknown(true)
+  .required()
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:125]: I think this `required` is redundant

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 2]
```diff
@@ -120,9 +120,28 @@ const ASSET_COMPONENT_SCHEME = Joi.object({
...
+type requestTypeComponent = {
+  value: {
+    requestType: ReferenceExpression
+    serviceDesk?: string
+  }
+}
+
+const REQUEST_TYPE_COMPONENT_SCHEME = Joi.object({
+  value: Joi.object({
+    requestType: Joi.required(),
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:138]: there is a mismatch between the scheme and the type, please add `serviceDesk` to the scheme

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 3]
```diff
@@ -343,36 +362,57 @@ const updateAutomationLabels = async (
...
+    })
+  }
+  if (config.fetch.enableJSM) {
+    const requestTypeComponents: requestTypeComponent[] = instance.value.components.filter(isRequestTypeComponent)
+    requestTypeComponents.forEach(component => {
+      try {
+        const requestType = component.value.requestType.value
+        component.value.serviceDesk = requestType.value.serviceDeskId
+      } catch (e) {
+        log.warn(`Failed to update detailed request type components for ${instance.elemID.getFullName()}`)
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:392]: 1. why do we need this try catch if we have a `schemeGuard`?
2. will the deployment be successful if we will fail here or [here](https://github.com/salto-io/salto/pull/5712/files#diff-5b83c3894d069ef363b848bb1aca51518f077ee83dfa2baa5e411c7be3e2ae4dR380)?

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 4]
```diff
@@ -343,36 +362,57 @@ const updateAutomationLabels = async (
...
   if (!instance.value.components) {
     return
   }
-  const assetsComponents: Component[] = instance.value.components.filter(isAssetComponent)
-  assetsComponents.forEach(component => {
-    delete component.value.schemaLabel
-    delete component.value.objectTypeLabel
-    delete component.value.workspaceId
-  })
+  if (config.fetch.enableJSM && (config.fetch.enableJsmExperimental || config.fetch.enableJSMPremium)) {
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:402]: we need to find a way to reduce those flags

File: packages/jira-adapter/src/reference_mapping.ts [hunk 5]
```diff
@@ -1323,6 +1323,11 @@ export const referencesRules: JiraFieldReferenceDefinition[] = [
     serializationStrategy: 'id',
     target: { type: OBJECT_SCHEMA_TYPE },
   },
+  {
```
Comment (shayc331) [inline packages/jira-adapter/src/reference_mapping.ts:1326]: do we want to add missing reference behavior? 
do we want to limit automation deployment when we don't have the `requestType`?

**Review by shayc331 — COMMENTED** [review]
💣 
looking great, add a few comments

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts (hunk 4 above)
Comment (idoamit8) [reply packages/jira-adapter/src/filters/automation/automation_deployment.ts:402]: Work in progress :) 
The customer success need to tell current customer to change `enableJsmExperimental` to `enableJSMPremium` and than we can drop this flag.

File: packages/jira-adapter/src/reference_mapping.ts (hunk 5 above)
Comment (idoamit8) [reply packages/jira-adapter/src/reference_mapping.ts:1326]: Yes, I will add missing reference behavior. Thanks

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 6]
```diff
@@ -343,36 +361,54 @@ const updateAutomationLabels = async (
...
-      const schema = component.value.schemaId.value
-      component.value.schemaLabel = schema.value.name
-      component.value.workspaceId = schema.value.workspaceId
-      if (component.value.objectTypeId !== undefined) {
-        const objectType = component.value.objectTypeId.value
-        component.value.objectTypeLabel = objectType.value.name
+  if (config.fetch.enableJSM && (config.fetch.enableJsmExperimental || config.fetch.enableJSMPremium)) {
+    const assetsComponents: AssetComponent[] = instance.value.components.filter(isAssetComponent)
+    assetsComponents.forEach(component => {
+      try {
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:371]: please remove the try-catch from here as well

File: packages/jira-adapter/src/filters/automation/automation_deployment.ts [hunk 7]
```diff
@@ -385,12 +384,9 @@ const modifyComponentsPreDeploy = (instance: InstanceElement, config: JiraConfig
...
     requestTypeComponents.forEach(component => {
-      try {
-        const requestType = component.value.requestType.value
-        component.value.serviceDesk = requestType.value.serviceDeskId
-      } catch (e) {
-        log.warn(`Failed to update detailed request type components for ${instance.elemID.getFullName()}`)
-      }
+      const requestType = component.value.requestType.value
+      component.value.serviceDesk = requestType.value.serviceDeskId
+      log.warn(`Failed to update detailed request type components for ${instance.elemID.getFullName()}`)
```
Comment (shayc331) [inline packages/jira-adapter/src/filters/automation/automation_deployment.ts:389]: please delete this log

**Review by shayc331 — COMMENTED** [review]
looking good! 
added 2 comments

**Review by shayc331 — APPROVED** [review]
💯

**Pull request merged**